src/components/ContactUs/DesktopVersion.tsx

```diff
@@ -42,17 +72,41 @@ export default function DesktopVersion() {
           </label>
           <textarea
             id="message"
+            name="message"
             placeholder="Hello, I'm getting in touch ..."
             className="border border-[#CEEDF4] w-full rounded-lg px-4 py-[22px] mb-9 h-[178px] resize-none outline-none placeholder-[#7E7979] font-montserrat text-base"
             required
           />
+          <ValidationError
+            prefix="Message"
+            field="message"
+            errors={state.errors}
+          />
           {/*//TODO: replace this button with the app button component */}
-          <button
-            type="submit"
-            className="block rounded-md p-2 my-2 w-[247px] h-14 mx-auto bg-[#EC0505] text-white font-bold font-montserrat"
-          >
-            Send Message
-          </button>
+
+          <ReCAPTCHA
+            sitekey={process.env.NEXT_PUBLIC_RECAPTCHA_SITE_KEY!}
+            onChange={handleRecaptchaResult}
+          />
+
+          {captchaComplete ? (
+            <button
+              type="submit"
+              disabled={state.submitting}
+              className="block rounded-md p-2 my-2 w-[247px] h-14 mx-auto bg-[#EC0505] text-white font-bold font-montserrat"
+            >
+              Send Message
+            </button>
+          ) : (
+            <button
+              type="submit"
+              disabled
+              className="block rounded-md p-2 my-2 w-[247px] h-14 mx-auto bg-[#EC0505] text-white font-bold font-montserrat"
+            >
+              Send Message
+            </button>
+          )}
```

Comment:
These two blocks are pretty much the same. The only difference is the `disabled` prop. We can simplify it to:

```tsx
<button
  type="submit"
  disabled={!captchaComplete || state.submitting}
  className="block rounded-md p-2 my-2 w-[247px] h-14 mx-auto bg-[#EC0505] text-white font-bold font-montserrat"
>
  Send Message
</button>
```

The same applies to the mobile version below.

Reply:
Nice catch. I'm shamelessly violating DRY and it's not good 😆